x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/TransportSubmitAsyncSearchAction.java

```diff
@@ -88,42 +85,34 @@ public void onResponse(AsyncSearchResponse searchResponse) {
                         // the task is still running and the user cannot wait more so we create
                         // a document for further retrieval
                         try {
-                            if (submitTask.isCancelled()) {
```

Comment:
I was wondering if it can still happen that we get here and submit is cancelled.

Reply:
I think that's ok, the cancelled may be delayed but we'll remove the document at the end (onFinalResponse checks if the task was cancelled).